### PR TITLE
Fixed sporadic in StressPipelineConfigChanges.GetMetricsDuringLoad

### PR DESCRIPTION
**Pull request by bstrzele**

### CODE REVIEW

File: src/test/ensemble_config_change_stress.cpp [hunk 1]
```diff
@@ -1164,8 +1164,17 @@ class StressPipelineConfigChanges : public TestWithTempDir {
     }
     void testCurrentRequestsMetric() {
         SPDLOG_INFO("{} start", __FUNCTION__);
-        checkMetricGreaterThan(METRIC_NAME_CURRENT_REQUESTS, 0);
-        checkMetricGreaterThan(METRIC_NAME_INFER_REQ_ACTIVE, 0);
+        bool current_reqests_pass, infer_req_active_pass;
```
Comment (dkalinowski) [inline src/test/ensemble_config_change_stress.cpp:1167]: Initialize with false

File: src/test/ensemble_config_change_stress.cpp [hunk 2]
```diff
@@ -1164,8 +1164,17 @@ class StressPipelineConfigChanges : public TestWithTempDir {
...
+        int retries = 3;
+        for (int i = 0; i < retries; i++) {
+            std::string metricOutput = manager.getMetricRegistry()->collect();
+            checkMetricGreaterThan(METRIC_NAME_CURRENT_REQUESTS, 0, metricOutput, current_reqests_pass);
+            checkMetricGreaterThan(METRIC_NAME_INFER_REQ_ACTIVE, 0, metricOutput, infer_req_active_pass);
+            if (current_reqests_pass && infer_req_active_pass)
+                break;
+        }
+        if (!current_reqests_pass || !infer_req_active_pass)
+            FAIL();
```
Comment (dkalinowski) [inline src/test/ensemble_config_change_stress.cpp:1177]: Can we add message to FAIL()?

```
FAIL() << "explain what failed";
```

File: src/test/ensemble_config_change_stress.cpp (hunk 1 above)
Comment (dkalinowski) [inline src/test/ensemble_config_change_stress.cpp:1167]:
```suggestion
        bool current_requests_pass, infer_req_active_pass;
```

**Comment by dkalinowski** [comment]
As @atobiszei  noted, we could also add debug log stating that "collection" was done to distinguish timing from serializing into prometheus format. In separate PR?

**Review by dkalinowski — APPROVED** [review]

**Review by dtrawins — APPROVED** [review]

**Pull request merged**